### PR TITLE
Test 3d secure auth bypassing web-mock 

### PR DESCRIPTION
**Pull request by WencesLlobet**

### CODE REVIEW

File: test/zebra/payment_intents_test.clj [hunk 1]
```diff
@@ -65,17 +67,29 @@
...
-      (is (= (:status payment-intent) "requires_action"))
-      (is (= (:confirmation_method payment-intent) "automatic"))
-      (is (= (:payment_method_types payment-intent) ["card"]))
-      (is (vector? (:payment_method_types payment-intent)))
-      (is (= (:amount payment-intent) 1234))
-      (is (= (:currency payment-intent) "gbp"))
-      (is (= (:payment_method payment-intent) (:id payment-method)))
-      (is (= (:type (:next_action payment-intent)) "redirect_to_url")))))
+        (do
+          (def driver (create-driver {:driver-type :firefox :driver-args ["--headless"]}))
```
Comment (tobyclemson) [inline test/zebra/payment_intents_test.clj]: This can go in the `let` above.

File: test/zebra/payment_intents_test.clj [hunk 2]
```diff
@@ -65,17 +67,29 @@
...
-      (is (= (:object payment-intent) "payment_intent"))
-      (is (= (:status payment-intent) "requires_action"))
-      (is (= (:confirmation_method payment-intent) "automatic"))
-      (is (= (:payment_method_types payment-intent) ["card"]))
-      (is (vector? (:payment_method_types payment-intent)))
-      (is (= (:amount payment-intent) 1234))
-      (is (= (:currency payment-intent) "gbp"))
-      (is (= (:payment_method payment-intent) (:id payment-method)))
-      (is (= (:type (:next_action payment-intent)) "redirect_to_url")))))
+        (do
```
Comment (tobyclemson) [inline test/zebra/payment_intents_test.clj]: There's no need for a `do` here. The body of a `let` is an implicit `do`.

File: test/zebra/payment_intents_test.clj [hunk 3]
```diff
@@ -65,17 +67,29 @@
...
+          (def driver (create-driver {:driver-type :firefox :driver-args ["--headless"]}))
+          (to driver (:url (:redirect_to_url (:next_action payment-intent))))
+          (wait-for-element driver :name "__privateStripeFrame4")
+          (iframe driver "__privateStripeFrame4")
+          (wait-for-element driver :name "stripe-challenge-frame")
+          (iframe driver "stripe-challenge-frame")
+          (send-keys (wait-for-element driver :xpath "//*[@id=\"test-source-authorize-3ds\"]") (. org.openqa.selenium.Keys ENTER)))
+          (Thread/sleep 500)
```
Comment (tobyclemson) [inline test/zebra/payment_intents_test.clj]: The strategy of using a firefox webdriver for this is going to make it difficult to run these tests in a deployment pipeline. Maybe look at using [htmlunit-driver](https://github.com/SeleniumHQ/htmlunit-driver).

**Review by tobyclemson — CHANGES_REQUESTED** [review]

File: test/zebra/payment_intents_test.clj (hunk 1 above)
Comment (WencesLlobet) [reply test/zebra/payment_intents_test.clj]: :+1:

File: test/zebra/payment_intents_test.clj (hunk 2 above)
Comment (WencesLlobet) [reply test/zebra/payment_intents_test.clj]: :+1:

File: test/zebra/payment_intents_test.clj (hunk 3 above)
Comment (WencesLlobet) [reply test/zebra/payment_intents_test.clj]: Thanks Toby, will take a look.

Comment (WencesLlobet) [reply test/zebra/payment_intents_test.clj]: PR is not doable because we encountered problems described in here:
https://stackoverflow.com/questions/20315330/how-to-overcome-an-htmlunit-scriptexception

TL;DR: HtmlUnit can break easily with javascript tweaks, even with minimization. Its is happening in the 3DS automating part

We tried:
- Different browsers versions in the library
- Deactivate javascript
- Deactivate image downloading
- Deactivate error messages

None of them worked for HtmlUnit and t seems to be the most robust "browser as a lib" tool.

We push the changes for error demo purposes
